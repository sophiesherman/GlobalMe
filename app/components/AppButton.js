import React from "react";
import { TouchableOpacity, StyleSheet, View } from "react-native";

import AppColors from "../config/AppColors";
import AppFonts from "../config/AppFonts";
import AppIcon from "./AppIcon";
import AppText from "./AppText";

// NAME: AppButton: Custom Component

// PURPOSE: Consistency for all buttons throughout the App

// PARAMS:
//  - color: default secondaryColor, color of the button
//  - onPress: function called when the button is pressed
//  - title: text written on the button
//  - width: default 180, width of the button
//  - height: default 60, height of the button
//  - icon: name of the material community icon to be used. See https://materialdesignicons.com/ for names

function AppButton({
  color = "secondaryColor",
  onPress,
  title,
  width = 180,
  height = 60,
  icon,
}) {
  return (
    <TouchableOpacity onPress={onPress}>
      <View
        style={[
          styles.button,
          { backgroundColor: AppColors[color], width: width, height: height },
        ]}
      >
        <AppText style={styles.text} font="subheading">
          {title}
        </AppText>
        {icon && [<AppIcon name="plus" size={50} color="white" />]}
      </View>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  button: {
    flexDirection: "row",
    borderRadius: 10,
    justifyContent: "center",
    alignItems: "center",
    borderColor: AppColors.white,
    shadowColor: '#171717',
    shadowOffset: {width: -2, height: 4},
    shadowOpacity: 0.2,
    shadowRadius: 3,
  },
  text: {
    fontSize: 22,
    color: AppColors.white,
    fontFamily: AppFonts.body,
  },
});

export default AppButton;
