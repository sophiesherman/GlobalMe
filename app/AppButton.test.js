import React from "react";
import renderer from "react-test-renderer";
import AppButton from "./components/AppButton";

test("AppButton renders correctly", () => {
  const tree = renderer.create(<AppButton />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("AppButton will render", () => {
  const json = renderer.create(<AppButton />).toJSON();
  expect(json.children[0].props.style[0].flexDirection).toBe("row");
  expect(json.children[0].props.style[0].borderRadius).toBe(10);
});

test("AppButton contains title", () => {
  const json = renderer
    .create(
      <AppButton
        title="Selected title here"
        onPress={() => console.log("Pressed")}
      />
    )
    .toJSON();
  expect(json.children.includes("Selected title here"));
  expect(json.props.style.opacity).toBe(1);
});
