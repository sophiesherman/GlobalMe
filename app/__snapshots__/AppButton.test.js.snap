// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppButton renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderBottomWidth": 2,
          "borderColor": "#FFF",
          "borderRadius": 10,
          "borderStartWidth": 2,
          "flexDirection": "row",
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "#4BAECE",
          "height": 60,
          "width": 180,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": undefined,
            "fontFamily": "Avenir Next",
          },
          Object {
            "color": "#FFF",
            "fontFamily": "Avenir Next",
            "fontSize": 22,
          },
        ]
      }
    >
       
       
    </Text>
  </View>
</View>
`;
